osu.Game/Screens/SelectV2/WedgeDifficultyDisplay.cs

```diff
@@ -0,0 +1,379 @@
+                                        {
+                                            new Dimension(),
+                                            new Dimension(GridSizeMode.Absolute, 30),
+                                            new Dimension(GridSizeMode.AutoSize),
+                                        },
+                                        Content = new[]
+                                        {
+                                            new[]
+                                            {
+                                                countStatisticsDisplay = new WedgeDifficultyStatisticsDisplay
+                                                {
+                                                    RelativeSizeAxes = Axes.X,
+                                                },
+                                                Empty(),
+                                                difficultyStatisticsDisplay = new AdjustableDifficultyStatisticsDisplay(autosize: true),
+                                            }
+                                        },
+                                    }
+                                },
+                            }
+                        }),
+                    }
+                },
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            beatmap.BindValueChanged(_ => updateDisplay());
+            ruleset.BindValueChanged(_ => updateDisplay());
+
+            mods.BindValueChanged(m =>
+            {
+                settingChangeTracker?.Dispose();
+
+                updateDifficultyStatistics();
+
+                settingChangeTracker = new ModSettingChangeTracker(m.NewValue);
+                settingChangeTracker.SettingChanged += _ => updateDifficultyStatistics();
+            });
+
+            updateDisplay();
+
+            displayedStars.BindValueChanged(_ => updateStars(), true);
+            FinishTransforms(true);
+        }
+
+        [Resolved]
+        private ILinkHandler? linkHandler { get; set; }
+
+        private void updateDisplay()
+        {
+            cancellationSource?.Cancel();
+            cancellationSource = new CancellationTokenSource();
+
+            computeStarDifficulty(cancellationSource.Token);
+
+            if (beatmap.IsDefault)
+                ratingAndNameContainer.FadeOut(300, Easing.OutQuint);
+            else
+            {
+                ratingAndNameContainer.FadeIn(300, Easing.OutQuint);
+                difficultyText.Text = beatmap.Value.BeatmapInfo.DifficultyName;
+                mapperLink.Action = () => linkHandler?.HandleLink(new LinkDetails(LinkAction.OpenUserProfile, beatmap.Value.Metadata.Author));
+                mapperText.Text = beatmap.Value.Metadata.Author.Username;
+            }
+
+            var playableBeatmap = beatmap.Value.GetPlayableBeatmap(ruleset.Value);
```

Comment:
Without doing a full review, I'll just mention that this is causing a beatmap load on the update thread